lang/cs/Org.Apache.REEF.Wake/Remote/Impl/TcpClientFactory.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using System.Net;
+using System.Net.Sockets;
+using Org.Apache.REEF.Tang.Annotations;
+using Org.Apache.REEF.Utilities.Logging;
+
+namespace Org.Apache.REEF.Wake.Remote.Impl
+{
+    /// <summary>
+    /// Default implementation of ITcpClientFactory.
+    /// Provides TcpClient for the remote endpoint with 
+    /// the specified retry logic for connection.
+    /// </summary>
+    public class TcpClientFactory : ITcpClientFactory
```

Comment:
I think name is ok...otherwise "TcpClientConnectionFactory" will be too long...since we pass endpoint any case in function call I feel it is sort of clear that connection is also established inside.